hudi-spark/src/main/java/org/apache/hudi/OverwriteWithLatestAvroPayload.java

```diff
@@ -61,8 +60,15 @@ public OverwriteWithLatestAvroPayload preCombine(OverwriteWithLatestAvroPayload
 
   @Override
   public Option<IndexedRecord> combineAndGetUpdateValue(IndexedRecord currentValue, Schema schema) throws IOException {
+
```

Comment:
if we change the codes in `combineAndGetUpdateValue()`, but row doesn't have `_hoodie_delete_marker` column, we need to `Option.of(genericRecord)` twice.
Should we put the codes in `getinsertvalue`? like this:
```java
  @Override
  public Option<IndexedRecord> combineAndGetUpdateValue(IndexedRecord currentValue, Schema schema) throws IOException {
    // combining strategy here trivially ignores currentValue on disk and writes this record
    return getInsertValue(schema);
  }

  @Override
  public Option<IndexedRecord> getInsertValue(Schema schema) throws IOException {
    GenericRecord baseRecord = HoodieAvroUtils.bytesToAvro(recordBytes, schema);
    Object deleteMarker = baseRecord.get("_hoodie_delete_marker");
    if (deleteMarker instanceof Boolean && (boolean) deleteMarker) {
      return Option.empty();
    }
    return Option.of(baseRecord);
  }
``

Reply:
@vinothchandar 
> Doing this in getInsertValue() means even inserts with the flag set will be deleted.. Not sure if this is intended behavior.. We only want to delete if updating and marker set?

If this is in a Kaapa architecture, it works. But if this is in a similar Lambda architecture, data should be rebuilt sometimes, it may will get whole data change logs by bulk insert.
Of course, this is just my assumption. Maybe our test cases haven't happen at present. If I think too much, and in fact it can't be found in actual cases, please ignore my review.